src/java/org/apache/cassandra/io/sstable/SSTableRewriter.java

```diff
@@ -87,24 +88,24 @@ public SSTableRewriter(ILifecycleTransaction transaction, long maxAge, long pree
         this.eagerWriterMetaRelease = eagerWriterMetaRelease;
     }
 
-    public static SSTableRewriter constructKeepingOriginals(ILifecycleTransaction transaction, boolean keepOriginals, long maxAge)
+    public static SSTableRewriter constructKeepingOriginals(ILifecycleTransaction transaction, SSTableFormat<?, ?> format, boolean keepOriginals, long maxAge)
     {
-        return new SSTableRewriter(transaction, maxAge, calculateOpenInterval(true), keepOriginals, true);
+        return new SSTableRewriter(transaction, maxAge, calculateOpenInterval(format, true), keepOriginals, true);
     }
 
-    public static SSTableRewriter constructWithoutEarlyOpening(ILifecycleTransaction transaction, boolean keepOriginals, long maxAge)
+    public static SSTableRewriter constructWithoutEarlyOpening(ILifecycleTransaction transaction, SSTableFormat<?, ?> format, boolean keepOriginals, long maxAge)
     {
-        return new SSTableRewriter(transaction, maxAge, calculateOpenInterval(false), keepOriginals, true);
+        return new SSTableRewriter(transaction, maxAge, calculateOpenInterval(format, false), keepOriginals, true);
     }
 
-    public static SSTableRewriter construct(ColumnFamilyStore cfs, ILifecycleTransaction transaction, boolean keepOriginals, long maxAge)
+    public static SSTableRewriter construct(ColumnFamilyStore cfs, SSTableFormat<?, ?> format, ILifecycleTransaction transaction, boolean keepOriginals, long maxAge)
     {
-        return new SSTableRewriter(transaction, maxAge, calculateOpenInterval(cfs.supportsEarlyOpen()), keepOriginals, true);
+        return new SSTableRewriter(transaction, maxAge, calculateOpenInterval(format, cfs.supportsEarlyOpen()), keepOriginals, true);
     }
 
-    private static long calculateOpenInterval(boolean shouldOpenEarly)
+    private static long calculateOpenInterval(SSTableFormat<?, ?> format, boolean shouldOpenEarly)
     {
-        long interval = DatabaseDescriptor.getSSTablePreemptiveOpenIntervalInMiB() * (1L << 20);
+        long interval = (long) format.getSSTableFormatValue(Option.SSTABLE_PREEMPTIVE_OPEN_INTERVAL) * (1L << 20);
```

Comment:
The format will not necessarily support this. To make it accessibly in a general way, define e.g. a `PreemptiveOpenConfiguration` interface that provides this value and make the two format implement it. If the given format implements that interface, get the value from it. If not, disable early open (i.e. use -1).